src/controllers/docs.controller.ts

```diff
@@ -0,0 +1,34 @@
+import { NextFunction, Request, Response, Router } from 'express';
+
+import redoc from 'redoc-express';
+
+import { Controller } from '../interfaces';
+
+export class DocsController implements Controller {
+  public basepath = '/docs';
+
+  private async docs(req: Request, res: Response, next: NextFunction) {
+    try { 
+
+      await redoc({
+        title: 'API Docs',
+        specUrl: '.../openapi.yaml'
+      });
+
+      res.status(204).end();
+    } catch (err) {
+      return next(err);
+    }
+  }
```

Comment:
I don't think we need this function on this controller. The `redoc-express` middleware can be used while setting up the route handler in `boot`. If you don't want to do this on that function, you can still use this `docs` function, but call the `redoc` function like you would call a middleware 👍 (with `req` and `res`). 
Their repo has this example:

```js
// serve your swagger.json file
app.get('/docs/swagger.json', (req, res) => {
  res.sendFile('swagger.json', { root: '.' });
});

// define title and specUrl location
// serve redoc
app.get(
  '/docs',
  redoc({
    title: 'API Docs',
    specUrl: '/docs/swagger.json'
  })
);
```

We don't want to return a `204`, but instead return the HTML page that gets you to visualize your open API spec 🙂 

If you have doubts doing this let us know 👍

Reply:
Hi, thanks for the review. I'm new to typescript and it's a little confusing. However, I have read your comments and tried to implement the changes. Let me know if this is what you expect.

```
export class DocsController implements Controller {

  public basepath = '/docs';

  public boot(): Router {
    const router = Router();
    
    router.get('/docs', (req, res) => {
      res.sendFile('.../openapi.yaml');
    });

    router.post(
      `${this.basepath}`,
      redoc({
        title: 'API Docs',
        specUrl: '.../openapi.yaml'
      })
    );

    
    return router;
  }
}
```

Reply:
@everly-gif Yeah, but `specUrl` should point to the `/docs/openapi.yaml` :) Rest is good. Please propagate changes to the PR.

Reply:
I have made the changes ! Thanks a lot for the help.